### ISSUE TITLE
unable to set expiration_policy.ttl to empty string

### ISSUE THREAD

**Issue by rltoscano** [terraform-google-modules/terraform-google-pubsub#187]
### TL;DR

In order to prevent a subscription from expiring, the TTL on the subscriptions expiration policy must be set to empty string (if set to null or not provided, then a default 31d expiration is used). However, when setting empty string on the `expiration_policy` field, the module sets `null` on the TTL field resulting in default 31d expiration instead of empty string.

### Expected behavior

Empty string is set on the subscription.expiration_policy.ttl field.

### Observed behavior

`null` is set on the subscription.expiration_policy.ttl field.

### Terraform Configuration

```hcl
module "mypubsub" {
  source     = "terraform-google-modules/pubsub/google"
  version    = "~> 6.0"
  project_id = var.project_id
  topic      = "mypubsub"
  push_subscriptions = [
    {
      name                       = "mysubscription"
      push_endpoint              = local.url
      x-goog-version             = "v1"
      oidc_service_account_email = local.identity
      expiration_policy          = ""
    }
  ]
}
```

### Terraform Version

```sh
1.5
```

### Additional information

I took a look at the module source code in this repo and don't understand why the empty string is turning into a null value. Something unexpected is happening around here: https://github.com/terraform-google-modules/terraform-google-pubsub/blob/c496f56a31f76e23f8414a03e02ad09aa2188f14/main.tf#L197-L203

**Comment by rltoscano** [comment]
Looks like the provider docs are misleading and the resource does not expire when ttl is set to null. I.e. it does not have to be set to empty string.
